### PR TITLE
[KYUUBI #1678] Fixed: The value of hasResultSet is always set to true in Kyuubi Server

### LINKED ISSUE #1678: [Bug] The value of hasResultSet is always set to true in Kyuubi Server
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

https://github.com/apache/incubator-kyuubi/blob/47533100f6f69b02a90985637a8fcc3f2cebb28c/kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala#L56-L73

`hasResultSet` is set to true by default in the `beforeRun` method. `hasResultSet` value can be obtained from `remoteOpHandle`, whether to consider setting the value in the `executeStatement` ？

Like this: 
```
 private def executeStatement(): Unit = {
    try {
      // We need to avoid executing query in sync mode, because there is no heartbeat mechanism
      // in thrift protocol, in sync mode, we cannot distinguish between long-run query and
      // engine crash without response before socket read timeout.
      _remoteOpHandle = client.executeStatement(statement, true, queryTimeout)
      setHasResultSet(_remoteOpHandle.isHasResultSet) 
    } catch onError()
  }
```

### Affects Version(s)

master/1.4.0

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by iodone**
Fixes #1678

### _Why are the changes needed?_

Updating `hasResultSet` state which is determined by the remote service

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
Seems test failed due to the flink memory issue, can you retrigger the GA @iodone thanks?

```java
FlinkOperationSuite:
Starting cluster.
Starting standalonesession daemon on host fv-az212-351.
Starting taskexecutor daemon on host fv-az212-351.
./build/mvn: line 118:  1840 Killed                  ${MVN_BIN} $MAVEN_CLI_OPTS "$@"
```

also cc @yanghua

**Comment by pan3793** [comment]
I'm finding the solution to reduce flink cluster memory usage.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala [hunk 1]
```diff
@@ -69,6 +69,7 @@ class ExecuteStatement(
       // in thrift protocol, in sync mode, we cannot distinguish between long-run query and
       // engine crash without response before socket read timeout.
       _remoteOpHandle = client.executeStatement(statement, true, queryTimeout)
+      setHasResultSet(_remoteOpHandle.isHasResultSet)
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala:72]: Shall we also update it according to TGetOperationStatusResp in waitStatementComplete

Comment (iodone) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala:72]: https://github.com/apache/incubator-kyuubi/blob/e0d9458144e68411f1dd62afb91efd1e1ef92cc5/kyuubi-common/src/main/scala/org/apache/kyuubi/service/ThriftBinaryFrontendService.scala#L415-L437

There is a unified configuration inside `GetOperationStatus` method, here we only need to change the operation's state

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**